chore: Biome lint

bmartel opened the pull request:
### Reason for change

To enable a consistent environment to work on and review frontend code in the MonoRepo, started by implementing Biome with recommended settings, and fixing errors along the way. This is a WIP branch to allow us to iteratively work at getting this settled in between other work.

bmartel committed:
chore: Setup biome and initial linting pass

bmartel committed:
rename component files to jsx

luarmr commented on web/apps/labelstudio/src/app/RootPage.js:
```diff
@@ -1,20 +1,22 @@
+      onSidebarToggle={(visible) =>
+        localStorage.setItem("sidebar-opened", visible)
+      }
+      onSidebarPin={(pinned) => localStorage.setItem("sidebar-pinned", pinned)}
```
Can we increase the number of columns to more than 80? This will significantly reduce the size of this PR.

luarmr commented on web/apps/labelstudio/src/app/AsyncPage/AsyncPage.js:
```diff
@@ -1,50 +1,60 @@
-
-      pageCache.set(url, html);
-      return html;
     }
```
Ths is outstanding

luarmr commented on web/apps/labelstudio/src/components/Form/Elements/Counter/Counter.js:
```diff
@@ -89,8 +96,8 @@ const Counter = ({label, className, validate, required, skip, labelProps, ...pro
-    if (type === 'increase') return increase();
-    else if (type === 'decrease') return decrease();
+    if (type === "increase") return increase();
+    if (type === "decrease") return decrease();
```
❤️

bmartel committed:
line width 120

bmartel committed:
reapply only safe on web src

robot-ci-heartex committed:
ci: Build frontend

Workflow run: https://github.com/HumanSignal/label-studio/actions/runs/8331912644

robot-ci-heartex commented:
This PR is stale because it has been open 45 days with no activity. Remove `stale` label or comment or this will be closed in 10 days.

bmartel commented:
Closing as not needed.
